### PR TITLE
Cs 1220 display stopped status on home page

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

File: src/SFA.DAS.ApprenticePortal.UnitTests/TestContext.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using SFA.DAS.ApprenticePortal.UnitTests.Hooks;
+
+namespace SFA.DAS.ApprenticePortal.UnitTests
+{
+    public class TestContext
```
Comment (benagain) [inline src/SFA.DAS.ApprenticePortal.UnitTests/TestContext.cs:5]: I quite like the pattern we use in APIM where we don't need the bindings - the bits of the context are lazily instantiated.

https://github.com/SkillsFundingAgency/das-apim-endpoints/blob/master/src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/TestContext.cs

I don't forsee the Portal having too many tests, so I don't think we need to change it now.  I'd like to use that pattern on new projects though.

File: src/SFA.DAS.ApprenticePortal.Web/Pages/Home.cshtml.cs [hunk 2]
```diff
@@ -2,24 +2,44 @@
...
+        private readonly ApprenticeshipService _apprenticeshipService;
         private readonly AuthenticatedUser _user;
         private readonly NotificationAccessor _notifications;
 
-        public string CourseName { get; set; } = null!;
-        public bool ApprenticeshipComplete { get; set; } = false;
+        public ApprenticeshipModel? CurrentApprenticeship { get; set; }
 
-        public HomeModel(IOuterApiClient client, AuthenticatedUser user, NotificationAccessor notifications)
+        public ApprenticeshipStatus Status
```
Comment (benagain) [inline src/SFA.DAS.ApprenticePortal.Web/Pages/Home.cshtml.cs:19]: Would it make sense to extract this to an extension method?

File: src/SFA.DAS.ApprenticePortal.Web/Pages/_HomeNotifications.cshtml [hunk 3]
```diff
@@ -31,3 +32,20 @@
         <p class="govuk-body">You now need to <a asp-external-section="ConfirmMyApprenticeship">confirm your apprenticeship details</a></p>
     </div>
 </div>
+
+@if (Model?.DisplayJustStoppedInfoMessage == true)
```
Comment (benagain) [inline src/SFA.DAS.ApprenticePortal.Web/Pages/_HomeNotifications.cshtml:36]: Do we have the `asp-show` tag helper in this project?  I think that may make the markup a touch more expressive.

File: src/SFA.DAS.ApprenticePortal.Web/Services/ApprenticeshipService.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.ApprenticePortal.Web.Models;
+using SFA.DAS.ApprenticePortal.Web.Services.OuterApi;
+
+namespace SFA.DAS.ApprenticePortal.Web.Services
+{
+    public class ApprenticeshipService
```
Comment (benagain) [inline src/SFA.DAS.ApprenticePortal.Web/Services/ApprenticeshipService.cs:9]: I wonder if we want an API that returns the Apprentice *and* the Apprenticeship, to save a HTTP call on every page load?

File: src/SFA.DAS.ApprenticePortal.UnitTests/FeaturesSteps/HomePageSteps.cs [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        [Given(@"the apprentice is authenticated")]
+        public void GivenTheApprenticeIsAuthenticated()
+        {
+            _context.Web.AuthoriseApprentice(_apprenticeId); 
+        }
+
+        [Given(@"there is a single incomplete apprenticeship")]
+        public void GivenThereIsASingleIncompleteApprenticeship()
+        {
+            _singleApprenticeship = _fixture.Build<Apprenticeship>().With(x => x.IsStopped, false)
```
Comment (benagain) [inline src/SFA.DAS.ApprenticePortal.UnitTests/FeaturesSteps/HomePageSteps.cs:39]: All of these setups that have to set `IsStopped` and `StoppedReceivedOn` to complimentary values suggests that I made the wrong decision to include `IsStopped` in the API.  I don't like have two separate fields that depend on being in step.

I don't think there's anything to do in this PR - maybe we can come back and remove that everywhere it's used at some point.

**Review by benagain — APPROVED** [review]

File: src/SFA.DAS.ApprenticePortal.Web/Models/ApprenticeshipModel.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
+﻿namespace SFA.DAS.ApprenticePortal.Web.Models
+{
+    public class ApprenticeshipModel
+    {
+        public string CourseName { get; set; } = "null!";
```
[the remark this answers is not in the document]
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticePortal.Web/Models/ApprenticeshipModel.cs:5]: there not.

File: src/SFA.DAS.ApprenticePortal.UnitTests/TestContext.cs (hunk 1 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticePortal.UnitTests/TestContext.cs:5]: I'm happy to use the pattern in APIM, but I do think there is value in implementing the Feature here as the menu page has several logic statements (at the moment) and currently we are not testing these flows

File: src/SFA.DAS.ApprenticePortal.Web/Pages/Home.cshtml.cs (hunk 2 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticePortal.Web/Pages/Home.cshtml.cs:19]: yes. agree

File: src/SFA.DAS.ApprenticePortal.Web/Pages/_HomeNotifications.cshtml (hunk 3 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticePortal.Web/Pages/_HomeNotifications.cshtml:36]: good idea

File: src/SFA.DAS.ApprenticePortal.UnitTests/FeaturesSteps/HomePageSteps.cs (hunk 5 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticePortal.UnitTests/FeaturesSteps/HomePageSteps.cs:39]: An alternative is to use an extension method (be nice to have an extension property)

File: src/SFA.DAS.ApprenticePortal.Web/Services/ApprenticeshipService.cs (hunk 4 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticePortal.Web/Services/ApprenticeshipService.cs:9]: I didn't want to fall into the trap of creating God objects, so it seems like a good idea to keep functionality in smaller classes with a defined propose.   Happy to discuss

Comment (benagain) [reply src/SFA.DAS.ApprenticePortal.Web/Services/ApprenticeshipService.cs:9]: I agree in terms of the classes in the project.  I was thinking (out load) about reducing the number of HTTP calls required.  Probably not important for this PR, let's have a chat about it in general.

File: src/SFA.DAS.ApprenticePortal.Web/Pages/_HomeNotifications.cshtml (hunk 3 above)
Comment (Pauljgraham) [reply src/SFA.DAS.ApprenticePortal.Web/Pages/_HomeNotifications.cshtml]: added the tag helper to this project

**Review by benagain — APPROVED** [review]

**Pull request merged**